ports/nrf/mphalport.c

```diff
@@ -77,3 +77,87 @@ void mp_hal_stdout_tx_strn_cooked(const char *str, mp_uint_t len) {
 void mp_hal_stdout_tx_str(const char *str) {
     mp_hal_stdout_tx_strn(str, strlen(str));
 }
+
+void mp_hal_delay_us(mp_uint_t us)
+{
+    register uint32_t delay __ASM ("r0") = us;
+    __ASM volatile (
+#ifdef NRF51
+            ".syntax unified\n"
+#endif
+        "1:\n"
+        " SUBS %0, %0, #1\n"
```

Comment:
It may be better to use named registers, like `subs %[delay_us], %[delay_us], #1`. It avoids forcing the `us` argument at `r0` (where it is anyway). I would say it is also more platform independent but I guess that doesn't really matter for such code...
See: http://www.ethernut.de/en/documents/arm-inline-asm.html